gen/service.go

```diff
@@ -280,3 +286,37 @@ func functionUnwrapResponse(g Generator, f *compile.FunctionSpec) (string, error
 			}
 		`, f)
 }
+
+func functionArgsEnveloper(g Generator, f *compile.FunctionSpec) error {
+	return g.DeclareFromTemplate(
+		`
+		<$wire := import "github.com/thriftrw/thriftrw-go/wire">
+		<$argType := printf "%v%v" (goCase .Name) "Args">
+		<$v := newVar "v">
+
+		func (<$v> *<$argType>) MethodName() string {
+			return "<.MethodName>"
+		}
+
+		func (<$v> *<$argType>) Type() <$wire>.EnvelopeType {
```

Comment:
Good point, perhaps we should rename the interface methods now, since `type` will show up pretty often in user types.

I'll call it `EnvelopeType` and put a TODO in to figure out how to deal with conflicts.